packages/tooltip/src/tooltipUtils.tsx

```diff
@@ -1,40 +1,83 @@
 import { css } from '@leafygreen-ui/emotion';
 import { Align, Justify } from '@leafygreen-ui/popover';
+import clamp from 'lodash/clamp';
 
-export function trianglePosition(
+export function notchPositionStyles(
   align: Align,
   justify: Justify,
   triggerRect: DOMRect | ClientRect | null,
 ) {
   if (!align || !justify || !triggerRect) {
-    return '';
+    return {
+      notchContainer: '',
+      notch: '',
+      tooltip: '',
+    };
   }
 
   const containerSize = 15;
   const notchSize = 8;
   const notchOverlap = -notchSize / 2;
 
   type Styles = 'left' | 'right' | 'top' | 'bottom' | 'margin';
-  const notchStyleObj: { [K in Styles]?: string } = {};
-  const containerStyleObj: { [K in Styles]?: string } = {};
+  const notchStyleObj: Partial<Record<Styles, string>> = {};
+  const containerStyleObj: Partial<Record<Styles, string>> = {};
+
+  // The bounds used to clamp the notchOffset value
+  const notchOffsetLowerBound = 5;
+  let notchOffsetUpperBound = notchOffsetLowerBound * 2;
+
+  // The un-clamped value that would exactly center the tooltip notch relative to the trigger.
+  let notchOffsetActual: number;
+
+  // The clamped value that makes a best-attempt to center the notch relative to the trigger,
+  // while also ensuring that the notch is positioned within the bounds of the tooltip itself,
+  // and still has the appearance of an alignment.
+  let notchOffset = 0;
+
+  // Boolean derived from the notchOffsetActual and notchOffsetLowerBound that determines if the trigger
+  // is small enough to make a transformation of the tooltip itself necessary.
+  let transformPosition: boolean;
+
+  // When the trigger is smaller than the minimum offset we require to position the notch over the trigger,
+  // we calculate a transformation to apply to the entire tooltip so that the notch centers on that element.
+  // This is particularly important for things like icons, and icon buttons where without this transformation,
+  // the tooltip's notch could be positioned entirely off of the trigger.
+  let tooltipOffsetTransform = '';
 
   switch (align) {
     case 'top':
     case 'bottom':
+      notchOffsetUpperBound = notchOffsetLowerBound * 3;
```

Comment:
why * 3?

Reply:
This number is sorta "magical". Quite frankly, this was going to be a far more magical number than this. I decided to tie it to an actual value that could create decent proportions instead.

Calculating the exact value this would need to be requires setting a ref on the Tooltip content wrapper, and getting the height / width of it. The problem was that the height / width was changing when the `open` prop was being set, causing the notch to lose its positioning in some cases before the tooltip transitioned out. This was the next best solution IMO, as in my testing it produced nearly the same result, but without the issue I mentioned.

Reply:
Wonder if we can inline any of this info into the code for later reference?

Reply:
I can add a comment to make this more clear